### PR TITLE
updating base docs to match close to kubernetes-template-project

### PR DESCRIPTION
**Pull request by chrislovecnm**
Updating some of the base does to better mirror our template project.

Closes https://github.com/kubernetes/kops/issues/1023

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1040)

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -187,7 +186,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright [yyyy] [name of copyright owner]
+   Copyright 2017 The Kubernetes Authors
```
Comment (kris-nova) [inline LICENSE]: Should we spanning years here?

```
Copyright 2016-2017 The Kubernetes Authors
```

CC @sarahnovotny

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,9 +1,31 @@
...
+
+1. Submit an issue describing your proposed change to the repo in question.
+1. The [repo owners](OWNERS) will respond to your issue promptly.
+1. If your proposed change is accepted, and you haven't already done so, sign a Contributor License Agreement (see details above).
+1. Fork the desired repo, develop and test your code changes.
+1. Submit a pull request.
+
+### Adding dependencies
+
+If your patch depends on new packages, add that package with [`godep`](https://github.com/tools/godep). Follow the [instructions to add a dependency](https://github.com/kubernetes/kubernetes/blob/master/docs/devel/development.md#godep-and-dependency-management).
```
Comment (kris-nova) [inline CONTRIBUTING.md]: If this is our official contrib guidelines, I think we need to consolidate all the information into one place.. 

So maybe we nuke the `adding_a_feauture` or at least rename it, and focus the content on something else? EG: `Development Lifecycle`? 

Also - is `godep` official? I honestly don't even know - I usually do everything by hand

Comment (chrislovecnm) [inline CONTRIBUTING.md]: I snagged these off of the example project we have in kubernetes.  We need to put in a link to our how to update vendor.

File: LICENSE (hunk 1 above)
Comment (chrislovecnm) [inline LICENSE]: We can do what is done here: https://github.com/kubernetes/kubernetes/blob/master/LICENSE#L190

**Comment by kris-nova** [comment]
@k8s-bot aws e2e test this

File: docs/development/release.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Release Process
```
Comment (kris-nova) [inline docs/development/release.md]: @chrislovecnm Can we nuke this in favor of https://github.com/kubernetes/kops/pull/1053

Comment (chrislovecnm) [inline docs/development/release.md]: I can remove the file.  Not on my mac right now.

File: LICENSE (hunk 1 above)
Comment (justinsb) [inline LICENSE]: We shouldn't change this, I thought... e.g. https://github.com/kubernetes/kubernetes/commit/d30db1f9a915aa95402e1190461469a1889d92be

File: CONTRIBUTING.md (hunk 2 above)
Comment (justinsb) [inline CONTRIBUTING.md]: Yeah, we'll want to remove all mention of Godeps before we let this near our project :-)

File: OWNERS [hunk 4]
```diff
@@ -0,0 +1,6 @@
+assignees:
+  - justinsb
+  - kris-nova
+  - chrislovecnm
+  - zmerlynn
+  - mikedanese
```
Comment (justinsb) [inline OWNERS]: I presume @mikedanese doesn't want to be listed as an assignee on kops?

File: docs/development/release.md (hunk 3 above)
Comment (justinsb) [inline docs/development/release.md]: Yeah, let's nuke this.  I don't think this process has been used by anyone - pure theory.

**Comment by chrislovecnm** [comment]
Updates done. cc: @kris-nova / @justinsb

**Comment by chrislovecnm** [comment]
@kris-nova knudge

**Comment by chrislovecnm** [comment]
Calling @kris-nova and @justinsb ... can I get a review :)

File: LICENSE [hunk 5]
```diff
@@ -187,7 +186,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright [yyyy] [name of copyright owner]
+   Copyright 2014 The Kubernetes Authors
```
Comment (justinsb) [inline LICENSE]: Can you revert this change?

**Comment by justinsb** [comment]
We should probably revert the change to apache license, but otherwise LGTM
